examples/delete_disks.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Show how to delete many disks created by the 'add_disks.py' example quickly.
+"""
+
+import time
+from contextlib import closing
+
+from ovirtsdk4 import types
+
+from helpers import common
+from helpers.common import progress
+from helpers.units import humansize, GiB
+
+parser = common.ArgumentParser(description="Delete disks created by the 'add_disks.py' example")
+
+parser.add_argument(
+    "--sd-name",
+    required=True,
+    help="Name of the storage domain.")
+
+parser.add_argument(
+    "--description",
+    default="Created by add_disks.py",
+    help="The description of the disks to be deleted.")
+
+args = parser.parse_args()
+
+progress("Connecting...")
+connection = common.create_connection(args)
+with closing(connection):
+    disks_service = connection.system_service().disks_service()
+    storage_domains_service = connection.system_service().storage_domains_service()
+    sd = storage_domains_service.list(search='name={}'.format(args.sd_name))[0]
+    sd_service = storage_domains_service.storage_domain_service(sd.id)
+    disks_service = sd_service.disks_service()
+    waiting = set()
+    count = 0
+
+    # StorageDomainDisksService.list has no 'search' parameter and ignores
+    # query={'name': 'spam'} so we have to do the filtering ourselves
+    disks = [disk for disk in disks_service.list() if disk.description == args.description]
+    for disk in disks:
+        count += 1
+        disk_service = disks_service.disk_service(disk.id)
+        waiting.add(disk.id)
+        disk_service.remove()
+
+        progress("Waiting for disk to be removed...")
+        while waiting:
+            time.sleep(1)
+            for disk_id in list(waiting):
+                if disk_id not in [disk.id for disk in sd_service.disks_service().list()]:
+                    progress("Disk %s/%s %s was removed" % (count, len(disks), disk_id))
+                    waiting.remove(disk_id)
```

Comment:
Listing all the disks in the domain to find if one disk was removed for every disk in the waiting list every second is extremely inefficient. What we want to instead is:

    while waiting:
        time.sleep(5)
        for disk in list(waiting):
            try:
                disks_service.disk_service(disk.id).get()
            except sdk.NotFoundError:
                waiting.discard(disk)
                progress(f"Removed {len(disks)-len(waiting)}/{len(disks)} disks")

Reply:
liked, ack